### PR TITLE
Make index and delete operation execute as single bulk item (backport of #21964)

### PR DESCRIPTION
**Pull request by areek**
Performance testing by @danielmitterdorfer revealed single
index/delete operations have similar performance (indexing
throughput) to equivalent single item bulk request.
This PR reduces the code paths to executing single write
operations, by reusing the logic in (shard) bulk action for
executing single operation as a single-item bulk request.

relates to #21964

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/index/IndexWithShadowReplicasIT.java [hunk 1]
```diff
@@ -86,6 +87,7 @@
 /**
  * Tests for indices that use shadow replicas and a shared filesystem
  */
+@LuceneTestCase.AwaitsFix(bugUrl = "fix this fails intermittently")
```
Comment (dakrone) [inline core/src/test/java/org/elasticsearch/index/IndexWithShadowReplicasIT.java]: I actually think these aren't needed any more, I think Boaz fixed the issue with these, they were removed from the `master` branch

File: core/src/test/java/org/elasticsearch/index/IndexWithShadowReplicasIT.java [hunk 2]
```diff
@@ -452,6 +454,7 @@ public void run() {
         assertHitCount(resp, numPhase1Docs + numPhase2Docs);
     }
 
+    @AwaitsFix(bugUrl = "uncaught exception")
```
Comment (dakrone) [inline core/src/test/java/org/elasticsearch/index/IndexWithShadowReplicasIT.java]: Same here

**Review by dakrone — APPROVED** [review]
LGTM, I left two comments about the `AwaitsFix`'d tests

File: core/src/test/java/org/elasticsearch/index/IndexWithShadowReplicasIT.java (hunk 1 above)
Comment (bleskes) [inline core/src/test/java/org/elasticsearch/index/IndexWithShadowReplicasIT.java]: yeah - this is the fix: https://github.com/elastic/elasticsearch/pull/22631/ which is already backported to 5.x

**Comment by areek** [comment]
thanks @dakrone for the review, I removed the await fixes in the tests and the CI is happy

**Pull request merged**